src/features/TakeNow/lib/useTakeNowValidation.tsx

```diff
@@ -0,0 +1,157 @@
+import { useState } from 'react';
+
+import { MultiInformantState } from '~/abstract/lib/types/multiInformant';
+import { useActivityByIdQuery } from '~/entities/activity';
+import { useAppletByIdQuery } from '~/entities/applet';
+import { useSubjectQuery } from '~/entities/subject';
+import { useUserState } from '~/entities/user/model/hooks';
+import { useWorkspaceAppletRespondent, useWorkspaceRolesQuery } from '~/entities/workspace';
+import { TakeNowParams } from '~/features/TakeNow/lib/TakeNowParams.types';
+import { WorkspaceRole } from '~/shared/api/types/workspace';
+import { useCustomTranslation } from '~/shared/utils';
+
+const TAKE_NOW_ROLES: WorkspaceRole[] = ['super_admin', 'owner', 'manager'];
+
+type TakeNowValidatedState = {
+  isLoading: boolean;
+  isError: boolean;
+  isSuccess: boolean;
+  error?: string | null;
+  data?: Required<MultiInformantState>;
+};
+
+export const useTakeNowValidation = ({
+  appletId,
+  subjectId: targetSubjectId,
+  startActivityOrFlow,
+  respondentId,
+}: TakeNowParams): TakeNowValidatedState => {
+  const [workspaceId, setWorkspaceId] = useState<string>('');
+
+  const {
+    isError: isSubjectError,
+    data: subjectData,
+    isLoading: isLoadingSubject,
+  } = useSubjectQuery(targetSubjectId);
+
+  const {
+    isError: isAppletError,
+    data: appletData,
+    isLoading: isLoadingApplet,
+  } = useAppletByIdQuery({ isPublic: false, appletId });
+
+  const { isError: isActivityError, isLoading: isLoadingActivity } = useActivityByIdQuery({
+    isPublic: false,
+    activityId: startActivityOrFlow,
+  });
+
+  const {
+    isError: isWorkspaceRolesError,
+    data: workspaceRolesData,
+    isLoading: isLoadingWorkspaceRoles,
+  } = useWorkspaceRolesQuery(workspaceId, {
+    appletIds: [appletId],
+  });
+
+  const {
+    isError: isRespondentError,
+    data: respondentData,
+    isLoading: isLoadingRespondent,
+  } = useWorkspaceAppletRespondent(workspaceId, appletId, respondentId);
+
+  const { t } = useCustomTranslation();
+  const { user } = useUserState();
+
+  const loadingState: TakeNowValidatedState = {
+    isLoading: true,
+    isError: false,
+    isSuccess: false,
+  };
+
+  const errorState = (error: string | null): TakeNowValidatedState => ({
+    isLoading: false,
+    isError: true,
+    isSuccess: false,
+    error,
+  });
+
+  if (respondentId !== user.id) {
+    return errorState(t<string>('takeNow.invalidRespondent'));
```

Comment:
I'm not seeing any TSC errors come up when these `<string>` type assertions are omitted from the file. Perhaps a relic?
```suggestion
    return errorState(t('takeNow.invalidRespondent'));
```

Reply:
Interesting. It was a TS error that made me add them in the first place, but now it's gone 🤷

Anyway, I removed them 👍